apps/sample-app/src/components/Calendar.tsx

```diff
@@ -1,21 +1,52 @@
 import './styles.css';
 
 import { Button } from '@fluentui/react-components';
-import { calendar } from '@microsoft/teams-js';
+import { app, calendar } from '@microsoft/teams-js';
 import React from 'react';
 
-export const Calendar: React.FunctionComponent = () => {
-  const handleCalendar = async (): Promise<void> => {
-    // create calendar paramaters
+import { MessageListItem } from './Emails';
+
+const teamsDeepLinkHost = 'teams.microsoft.com';
+const teamsDeepLinkProtocol = 'https';
+const teamsDeepLinkAttendeesUrlParameterName = 'attendees';
+const teamsDeepLinkUrlPathForCalendar = '/l/meeting/new';
+
+export const handleNewMail = async (): Promise<void> => {
+  if (!calendar.isSupported()) {
+    const calendarParams: calendar.ComposeMeetingParams = {
+      attendees: ['emailAddress@microsoft.com'],
+    };
+    const attendeeSearchParameter =
+      calendarParams.attendees === undefined
+        ? ''
+        : `${teamsDeepLinkAttendeesUrlParameterName}=` +
+          calendarParams.attendees.map(attendee => encodeURIComponent(attendee)).join(',');
+
+    const deepLinkUrl = `${teamsDeepLinkProtocol}://${teamsDeepLinkHost}${teamsDeepLinkUrlPathForCalendar}?${attendeeSearchParameter}`;
+    app.openLink(deepLinkUrl);
+  } else {
     const calendarParams: calendar.ComposeMeetingParams = {
-      attendees: ['t-abarthakur@microsoft.com'],
+      attendees: ['emailAdd@microsoft.com'],
     };
-    // send to 'compose meeting'
     await calendar.composeMeeting(calendarParams);
-  };
+  }
+};
+
+export const handleOpenMail = async (e: MessageListItem): Promise<void> => {
+  if (!calendar.isSupported()) {
+    alert('open mail item is not supported');
```

Comment:
I think it could be a bit confusing for the devs looking to the app as the implementation example since it's saying 'opening mail item is not supported' when calendar is not rather than when mail is not supported- mail has an API called `openMailItem`. I think it would be clearer to say something like 'calendar is not supported- could not open this calendar item.'